front/src/applications/rollingStockEditor/components/RollingStockEditorCurves.tsx

```diff
@@ -0,0 +1,146 @@
+import RollingStockCurve from 'common/RollingStockSelector/RollingStockCurves';
+import { RollingStock } from 'common/api/osrdEditoastApi';
+import { isEmpty } from 'lodash';
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import Spreadsheet, { CellBase, Matrix, createEmptyMatrix } from 'react-spreadsheet';
+
+type Curve = {
+  max_efforts: number[];
+  speeds: number[];
+};
+
+export default function RollingStockEditorCurves(props: {
+  data?: RollingStock;
+  currentRsEffortCurve: RollingStock['effort_curves'];
+  setCurrentRsEffortCurve: Dispatch<SetStateAction<RollingStock['effort_curves']>>;
+}) {
+  const { data, currentRsEffortCurve, setCurrentRsEffortCurve } = props;
+
+  const { t } = useTranslation('rollingstock');
+  const selectedMode = currentRsEffortCurve ? Object.keys(currentRsEffortCurve.modes)[0] : ''; // 1er mode par défaut.
+  const EMPTY_MATRIX = createEmptyMatrix<CellBase<string>>(8, 2);
+
+  // TODO: recup les listes avec filtres pour multi courbes
+
+  const defaultCurve = (
+    currentEffortCurve: RollingStock['effort_curves']
+  ): Matrix<CellBase<string>> => {
+    const speedsList = currentEffortCurve.modes[`${selectedMode}`].curves[0].curve?.speeds;
+    const effortsList = currentEffortCurve.modes[`${selectedMode}`].curves[0].curve?.max_efforts;
+
+    const filledMatrix: (
+      | {
+          value: string;
+        }
+      | undefined
+    )[][] =
+      speedsList !== undefined && effortsList !== undefined
+        ? effortsList.map((effort, index) => [
+            { value: (speedsList[index] * 3.6).toString() },
+            { value: (effort / 10).toString() },
+          ])
+        : [];
+
+    if (isEmpty(filledMatrix)) return EMPTY_MATRIX;
+
+    return !isEmpty(filledMatrix) && filledMatrix.length > 8
+      ? filledMatrix
+      : filledMatrix.concat(createEmptyMatrix<CellBase<string>>(8 - filledMatrix.length, 2));
+  };
+
+  const [curves, setCurves] = useState<Matrix<CellBase<string>>>(
+    defaultCurve(currentRsEffortCurve)
+  );
+
+  const updateSpreadsheet = (e: Matrix<{ value: string }>) => {
+    const emptyRow = [undefined, undefined];
+    const sheetValues = e.filter(
+      (step) => step[0]?.value !== undefined || step[1]?.value !== undefined
+    );
+
+    if (!isEmpty(sheetValues) && sheetValues.length >= 8) {
+      sheetValues.push(emptyRow);
+      setCurves(sheetValues);
+    } else if (!isEmpty(sheetValues) && sheetValues.length < 8) {
+      const fillingRows = createEmptyMatrix<CellBase<string>>(8 - sheetValues.length, 2);
+      setCurves(sheetValues.concat(fillingRows));
+    } else {
+      setCurves(defaultCurve(currentRsEffortCurve));
+    }
+  };
+
+  const parseCurve = (e: Matrix<{ value: string }>) => {
+    const sheetValues = e
+      .filter(
+        (step) =>
+          step[0]?.value !== (undefined && /^\s+$/) && step[1]?.value !== (undefined && /^\s+$/)
```

Comment:
what are trying to do here ? `(undefined && /^\s+$/)`

`(undefined && /^\s+$/)` always evaluates as `undefined`

Reply:
Done :)